File: golang/数组/小于当前数字的数字.go
package main

func smallerNumbersThanCurrent(nums []int) []int {
	n := len(nums)
	res := make([]int, 0)

	for i := 0; i < n; i++ {
		sum := 0
		for j := 0; j < n; j++ {
			if nums[i] > nums[j] {
				sum++
			}
		}

		res = append(res, sum)
	}

	return res
}

File: python/数组/小于当前数字的数字.py
def smallerNumbersThanCurrent(nums):
    n = len(nums)
    res = [0] * n
    for i in range(n):
        sum = 0
        for j in range(n):
            if nums[i] > nums[j]:
                sum += 1

        res[i] = sum

    return res


nums = [8, 1, 2, 2, 4]
print(smallerNumbersThanCurrent(nums))
